apps/condo/domains/billing/schema/BillingAccount.test.js

```diff
@@ -2,404 +2,467 @@
-            const [context] = await createTestBillingIntegrationOrganizationContext(managerUserClient, organization, integration)
-            const [property] = await createTestBillingProperty(managerUserClient, context)
-            const [billingAccount] = await createTestBillingAccount(managerUserClient, context, property)
-
-            await expectToThrowAccessDeniedErrorToObj(async () => {
-                await BillingAccount.delete(managerUserClient, billingAccount.id)
+            test('Can be same if context is different', async () => {
+                const [account] = await createTestBillingAccount(admin, context, property, {
+                    globalId,
+                })
+                expect(account).toBeDefined()
+                expect(account).toHaveProperty('globalId', globalId)
+                const [anotherAccount] = await createTestBillingAccount(admin, anotherContext, anotherProperty, {
+                    globalId,
+                })
+                expect(anotherAccount).toBeDefined()
+                expect(anotherAccount).toHaveProperty('globalId', globalId)
             })
-        })
-
-        test('cannot be deleted by user', async () => {
-            const admin = await makeLoggedInAdminClient()
-            const { context } = await makeContextWithOrganizationAndIntegrationAsAdmin()
-            const [property] = await createTestBillingProperty(admin, context)
-            const [billingAccount] = await createTestBillingAccount(admin, context, property)
-            const user = await makeClientWithNewRegisteredAndLoggedInUser()
-
-            await expectToThrowAccessDeniedErrorToObj(async () => {
-                await BillingAccount.delete(user, billingAccount.id)
-            })
-        })
-
-        test('cannot be deleted by anonymous', async () => {
-            const admin = await makeLoggedInAdminClient()
-            const { context } = await makeContextWithOrganizationAndIntegrationAsAdmin()
-            const [property] = await createTestBillingProperty(admin, context)
-            const [billingAccount] = await createTestBillingAccount(admin, context, property)
-            const client = await makeClient()
-
-            await expectToThrowAccessDeniedErrorToObj(async () => {
-                await BillingAccount.delete(client, billingAccount.id)
+            test('Must be unique within the same context', async () => {
+                const [account] = await createTestBillingAccount(admin, context, property, {
+                    globalId,
+                })
+                expect(account).toBeDefined()
+                expect(account).toHaveProperty('globalId', globalId)
+                await expectToThrowInternalError(async () => {
+                    await createTestBillingAccount(admin, context, property, {
+                        globalId,
+                    })
+                }, 'billingAccount_unique_context_globalId')
```

Comment:
it's better use whole message

Reply:
What's the reason for that? This prefix + name of test is enough to find a place of usage. And collision is almost impossible. Since you're specking this message manually.